packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -764,6 +767,34 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     detachFiber(current);
   }
 
+  function commitProfileWork(finishedWork: Fiber): void {
+    switch (finishedWork.tag) {
+      case Profiler: {
+        if (enableProfileModeMetrics) {
+          finishedWork.pendingProps.callback.call(
+            null,
+            finishedWork.pendingProps.id,
+            finishedWork.alternate === null ? 'mount' : 'update',
+            finishedWork.stateNode,
+            finishedWork.treeBaseTime,
+          );
+        }
+
+        // Reset actualTime after successful commit.
+        // By default, we append to this time to account for errors and pauses.
+        finishedWork.stateNode = 0;
```

Comment:
I think Dan ran some performance tests that showed the `stateNode` should always use an object or null, so the hidden class is monomorphic, for performance reasons.

Reply:
Oh? cc @gaearon

Reply:
It's not ideal but since we're already doing this for context I guess this doesn't make it worse.

Reply:
It's easy enough to add a wrapper object if that would be better in any way. (I actually had one earlier and removed it last minute.)